boards/xtensa/esp32_devkitc_wroom/doc/index.rst

```diff
@@ -0,0 +1,203 @@
+.. _esp32_devkitc_wroom:
+
+ESP32-DEVKITC-WROOM
+###################
```

Comment:
I have considered that approach at the beginning but since that change is tightly bound with the `soc/xtensa/espressif_esp32` and hasn't much meaning without each other, I choose to merge this in the single PR.